src/applications/caregivers/definitions/UIDefinitions/sharedUI.js

```diff
@@ -0,0 +1,123 @@
+import { createUSAStateLabels } from 'platform/forms-system/src/js/helpers';
+import { states } from 'platform/forms/address';
+import { validateSSNIsUnique } from 'applications/caregivers/helpers';
+import { VeteranSSNInfo } from 'applications/caregivers/components/AdditionalInfo';
+
+const stateLabels = createUSAStateLabels(states);
+
+export const emailUI = label => email(`${label}  email address`);
+export const dateOfBirthUI = label =>
+  currentOrPastDateUI(`${label}  date of birth`);
+
+export const addressUI = address.uiSchema(' ', false);
+
+export const primaryPhoneNumberUI = label =>
+  phoneUI(`${label}  primary telephone number (including area code)`);
+
+export const alternativePhoneNumberUI = label =>
+  phoneUI(`${label}  alternate telephone number (including area code)`);
+
+export const vetRelationshipUI = label => ({
+  'ui:title': `What is the ${label}  relationship to the Veteran?`,
+});
+
+export const genderUI = label => ({
+  'ui:title': `${label}  sex`,
+  'ui:widget': 'radio',
+  'ui:options': { labels: { F: 'Female', M: 'Male' } },
+});
+
+export const fullNameUI = label => ({
+  first: {
+    'ui:title': `${label}  first name`,
+    'ui:errorMessages': {
+      required: `Please enter ${label}  first name`,
+    },
+  },
+  last: {
+    'ui:title': `${label}  last name`,
+    'ui:errorMessages': {
+      required: `Please enter ${label}  last name`,
+    },
+  },
+  middle: {
+    'ui:title': `${label}  middle name`,
+  },
+});
+
+export const ssnUI = label => ({
+  ...platformSsnUI,
+  'ui:title': `${label}  Social Security number or tax identification number`,
+  'ui:description': label === 'Veteran\u2019s' && VeteranSSNInfo(),
+  'ui:options': {
+    widgetClassNames: 'usa-input-medium',
+  },
+  'ui:errorMessages': {
+    pattern:
+      'Please enter a valid Social Security or tax identification number',
+    required: 'Please enter a Social Security or tax identification number',
+  },
+  'ui:validations': [
+    ...platformSsnUI['ui:validations'],
+    {
+      validator: (errors, fieldData, formData) => {
+        validateSSNIsUnique(errors, formData);
+      },
+    },
+  ],
+});
+
+export const confirmationEmailUI = (label, dataConstant) => ({
+  'ui:title': `Re-enter ${label}  email address`,
+  'ui:widget': 'email',
+  'ui:required': formData => !!formData[dataConstant],
+  'ui:validations': [
+    {
+      validator: (errors, fieldData, formData) => {
+        const doesEmailMatch = () =>
+          formData[dataConstant] === formData[`view:${dataConstant}`];
+
+        if (!doesEmailMatch()) {
+          errors.addError(
+            'This email does not match your previously entered email',
+          );
+        }
+      },
+    },
+  ],
+});
+
+export const addressWithoutCountryUI = label => ({
+  'ui:title': ' ',
+  'ui:order': ['street', 'street2', 'city', 'state', 'postalCode'],
+  street: {
+    'ui:title': `${label}  current street address`,
```

Comment:
Not a big deal, but it looks like every `${label}` in this file is followed by two spaces.

Reply:
Awesome catch would not have caught this one in a million years! 🚀

Reply:
In the Engineering all-hands on Friday, I'll talk about userscripts... check this one out - https://github.com/Mottie/GitHub-userscripts/wiki/GitHub-code-show-whitespace